lib/serialization/sequenceSpec.ts

```diff
@@ -1,24 +1,41 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 import { PropertyPath } from "./propertyPath";
-import { SerializationOptions, failDeserializeTypeCheck, failSerializeTypeCheck, resolveTypeSpec } from "./serializationOptions";
+import { SerializationOptions, failDeserializeTypeCheck, failSerializeTypeCheck, resolveTypeSpec, SerializationOutputType } from "./serializationOptions";
 import { TypeSpec } from "./typeSpec";
 
 export interface SequenceTypeSpec<TSerializedElement, TDeserializedElement> extends TypeSpec<TSerializedElement[], TDeserializedElement[]> {
   /**
    * The TypeSpec that defines each element in this SequenceTypeSpec.
    */
   elementSpec: TypeSpec<TSerializedElement, TDeserializedElement> | string;
+
+  /**
+   * The element name of the list elements. Only used when this is a root list.
+   */
+  xmlElementName?: string;
+}
+
+export interface SequenceSpecOptions {
+  /**
+   * The element name of the root element. Only used when this is a root list.
+   */
+  xmlRootName: string;
+
+  /**
+   * The element name of the sequence elements. Only used when this is a root list.
```

Comment:
These comments that say `Only used when this is a root list.` seem to be telling me that they belong on a different kind of `TypeSpec`. I think we should create a new `TypeSpec` named `RootSequenceTypeSpec` that extends `SequenceTypeSpec` and includes these extra properties. That way the extra properties can be required and then they can go at the front of the creator function's (`rootSequenceSpec()`) parameter list.
Does that sound reasonable to you?

Reply:
Talked this through. It's fine to just keep this the way it is.